src/lucky_record/field.cr

```diff
@@ -12,7 +12,8 @@ class LuckyRecord::Field(T)
     @param || value.to_s
   end
 
-  def add_error(message : String)
+  def add_error(message : Object = "is invalid")
+    message = message.call(@name.to_s, @value.to_s) if message.responds_to?(:call)
```

Comment:
`responds_to` makes it less clear what this object needs and could potentially lead to errors. There is a not super well documented feature that we can use to make it more clear what this accepts without relying on `responds_to?`

When a class includes a module, you can restrict it by that type. So I think we can do something like this

```crystal
module LuckyRecord::CallableErrorMessage
  abstract def call(name, value)
end

class CallableMessage
  include LuckyRecord::CallableErrorMessage
end

# Then `add_error` changes to:
def add_error(message : String)
# And adds an additional one
def add_error(message : (Proc | LuckyRecord::CallableErrorMessage))
```

Reply:
I think I like that. It makes it more clear the requirement for name and value.I have to admit I tried hacking together ways to provide some flexibility providing a proc or callable with 0 args, 1 arg and 2 args... but it was gnarly. I'm not sure if there's a way to do this properly in